api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -288,4 +288,17 @@ message GrpcJsonTranscoder {
   //
   // If unset, the current stream buffer size is used.
   google.protobuf.UInt32Value max_response_body_size = 16 [(validate.rules).uint32 = {gt: 0}];
+
+  // If true, query parameters that cannot be mapped to a corresponding
+  // protobuf field are captured in an HttpBody extension of UnknownVariableBindings.
+  // This overrides ``ignore_unknown_query_parameters`` and ``reject_unknown_query_parameters``.
+  bool capture_unknown_query_parameters = 17;
```

Comment:
IMO rather than override we should reject a config which suggests doing 2 different things to unknown config params.

Reply:
That makes a transition to this new behavior difficult for us, because config and binary can be pushed separately and we're currently operating on a patch that does the new behavior with the old config.

The transition plan for me is to push a new config that specifies *both* the things, then push the new binary, thereby rolling onto the non-patch implementation of the behavior.

I'd be much more comfortable if I could make this change in a followup if that would be okay?

Reply:
On second thought, maybe it makes more sense to bind this with a different extension with a better API, have the patch *and* the new one both operating during the transition with the client preferring the new encoding, and that way I can do an API that isn't as bad as the original one.

Reply:
I think API changes which make previously valid config invalid are disallowed unfortunately. 
Options would be to have a runtime guard default-false for the allow behavior, and you could flip it internally, or just allow it and I will put up with it with mild crankiness as I have in the myriad other instances I've asked folks to not allow for confusing behavior and the harsh realities of production have made it impossible =P

As a compromise how about at least logging a warning below if the prior config knob is ignored?

Reply:
I started to do that, then realized it doesn't make sense - "warning, your ignore directive is being ignored in favor of another setting that does still ignore it in the way you meant".

So instead I made the protodoc less confusing about what overrides what; capture doesn't really override ignore, it just also does ignore whether you set both or not. The only thing it really overrides is the same thing ignore also overrides, and that wasn't documented on ignore, it was only documented on the overridden field, so I've made the documentation for the new field match that.

Reply:
well in that case there should be a warning for both ignore and send iff strict but there's prior art for not warning about that so I'm just going to declare myself an api grinch and LGTM this.